src/modeler/AutoRest.Swagger/Validation/NonAppJsonTypeWarning.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using AutoRest.Core.Logging;
+using AutoRest.Core.Properties;
+using AutoRest.Core.Validation;
+
+namespace AutoRest.Swagger.Validation
+{
+    public class NonAppJsonTypeWarning : TypedRule<string>
+    {
+        private const string AppJsonType = "application/json";
+        /// <summary>
+        /// This rule passes if the entity contains application/json type 
+        /// </summary>
+        /// <param name="entity"></param>
+        /// <returns></returns>
+        public override bool IsValid(string entity)
+            => (entity.ToLowerInvariant().StartsWith(AppJsonType));
```

Comment:
`StartsWith` ? 

Hmm. that's probably a good idea, good catch. I can't see a case where we shouldn't do that, but there may be one day.